src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -874,6 +874,18 @@ void ConnectionHandlerImpl::CreateWebEngineDevice() {
   transport_manager_.CreateWebEngineDevice();
 }
 
+bool ConnectionHandlerImpl::IsAppPresentInConnections(
+    const uint32_t connection_key) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  transport_manager::ConnectionUID connection_handle = 0;
+  uint8_t session_id = 0;
+  PairFromKey(connection_key, &connection_handle, &session_id);
+  const auto connection_uid = ConnectionUIDFromHandle(connection_handle);
+
+  return (nullptr != GetPrimaryConnection(connection_handle));
```

Comment:
`GetPrimaryConnection` is not thread-safe so needs to synchronize access to the connections list here.

Reply:
@mvorobio added lock